### DISCUSSION TITLE
Redis pub/sub channel listener should execute messages parallely.

### DISCUSSION THREAD

**Discussion by Prashant-Thorat** [redis/lettuce#2018] [Q&A]
Hi for my chatting application I'm planning to use the redis pub/sub for server scalability. wanted to use Redis pub/sub for broadcasting messages between servers. 

`      
String redisUrl = "redis://" + host + ":" + port;
redisClient = RedisClient.create(RedisURI.create(redisUrl));
syncPool = redisClient.connect();
isRunning = true;

StatefulRedisPubSubConnection<String, String> connection = redisClient.connectPubSub();
RedisPubSubAsyncCommands<String, String> async = connection.async();

async.subscribe("messages");
connection.addListener(new RedisMessageHandler());
`

RedisMessageHandler is processing messages in sequential manner(i.e. one after another). but I wanted to send received message in parallel.

**Comment by mp911de** [comment]
Redis itself is single-threaded and the connection protocol is sequential. You can still offload work to a `Executor` from your listener to unblock the event loop.

**Reply by Prashant-Thorat** [reply]
@mp911de thanks for your response.
